lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Random.hs

```diff
@@ -314,12 +394,12 @@ decodeDerPath = decodeListIndef CBOR.decodeWord32 >>= \case
 
 -- | Derive a symmetric key for encrypting and authenticating the address
 -- derivation path.
-hdPassphrase :: Key 'RootK XPub -> Passphrase "address-der"
-hdPassphrase (Key rootXPub) = Passphrase $
+hdPassphrase :: Key 'RootK XPrv -> Passphrase "address-der"
+hdPassphrase masterKey = Passphrase $
     PBKDF2.generate
     (PBKDF2.prfHMAC SHA512)
     (PBKDF2.Parameters 500 32)
-    (unXPub rootXPub)
+    (unXPub . toXPub . getKey $ masterKey)
```

Comment:
why to use `toXPub` if we have : 
```
publicKey
    :: Key level XPrv
    -> Key level XPub
publicKey (Key k) =
    Key (toXPub k)
```
in `Cardano.Wallet.Primitive.AddressDerivation` ?

Reply:
ok, I have only now detected that `publicKey` does not take form like this anymore